frozen_mutation.cc

```diff
@@ -172,6 +172,10 @@ stop_iteration streamed_mutation_freezer::consume(clustering_row&& cr) {
 }
 
 stop_iteration streamed_mutation_freezer::consume(range_tombstone&& rt) {
+    if (_reversed) {
+        // undo reversing done for the native reversed format, coordinator still uses old reversing format
```

Comment:
This comment is kinda out of place: it mentions "coordinator", so it basically leaks knowledge of the outer layer (the user of the `streamed_mutation_freezer` abstraction) into the abstraction. But what if someone else wanted to use this class? It's pretty generic and potentially useful in many contexts. We should not assume that the abstraction is used in exactly one place, in exactly one context.

Instead, the interface of the class should specify that the freezer converts from "old reversing format" to "native reversing format" (whatever those things mean) in a comment. There we could give the coordinator as an example user who depends on this behavior.

Reply:
Fix: the freezer converts form "native reversing format" to "old reversing format" (in particular it assumes the input is in "native reversing format")

Reply:
I can move this reversing to the `reconcilable_result_builder` which **is** specific to read-repair.